### ISSUE TITLE
Sort messages in conversation with the newest at the top

### ISSUE THREAD

**Issue by reaseno** [thunderbird-conversations/thunderbird-conversations#1900]
How can I sort messages in the messages panel so that the newest messages are at the top. Exactly in the reverse order of the current standard. I have not been able to find a setting for this unfortunately.

If it is not possible, I would like to make a feature request for it.

**Comment by Standard8** [comment]
[As per the FAQ](https://github.com/thunderbird-conversations/thunderbird-conversations/wiki/FAQ#message-display), this generally hasn't been done because of the additional complexity that it would create in the code.

I think it might be a little more viable now with refactors of the code base but I would still be concerned that it would be difficult to support well (e.g. especially when inserting messages into the currently displayed conversation - could cause unwanted scrolling etc).

For me, this is a low priority compared to all the other issues and feature requests that come in.

**Comment by godzfire** [comment]
I'm shocked that there hasn't been more responses to this, but not being able to sort and see most recent replies at the top like Outlook is an absolute deal breaker for me. It seems counterintuitive you have to scroll down large email conversations just to see the most recent email.

**Comment by Vigilans** [comment]
> I'm shocked that there hasn't been more responses to this, but not being able to sort and see most recent replies at the top like Outlook is an absolute deal breaker for me. It seems counterintuitive you have to scroll down large email conversations just to see the most recent email.

Yes, this just prevents me from adopting this extension in the first place despite all other features and merits.

**Comment by godzfire** [comment]
I am hoping this can be re-evaluated to inclusion. It's stated "For me, this is a low priority compared to all the other issues and feature requests that come in.", however my previous reply in this thread got 11 thumbs up by fellow members feeling the same way.

This feature is something that is highly wanted.

**Comment by Standard8** [comment]
> This feature is something that is highly wanted.

I do understand that some people want this feature. A lot more people have signalled they want dark mode, some people want encrypted email working again (a major regression of the last Thunderbird upgrade).

As I stated before, the UX for this feature is not necessarily trivial to implement and needs thinking about and careful testing.

Unfortunately I barely get enough time at the moment to keep the add-on going, let alone think about new features. When I said this was low priority, I'm just being honest with you that I simply don't have time for it.

I'd be happy for someone to take this on, but of course, that would need to be someone that can write the appropriate code.

**Comment by philia897** [comment]
@Standard8 

I've made a simple fix to this issue (See in this pull request #2165 ). It simply reverse the order of messages without changing other things. This feature can be enabled in options.

But in this way, the quick reply function does not work then, as it will reply to the earliest email.

Looks like:

![reverse](https://github.com/user-attachments/assets/89895677-032f-4eab-a987-31f63843b622)

**Comment by lukas-remis** [comment]
Has this been added to the addon? I don't see any option in the settings to reverse the conversation order.

**Comment by Standard8** [comment]
> Has this been added to the addon? I don't see any option in the settings to reverse the conversation order.

The issue says "Open" at the top. When issues are completed they'll be marked with "Closed".

**Comment by godzfire** [comment]
@Standard8 status update on this? This has been requested for 4 years now. It appears that [Symplify62](https://github.com/Symplify62) added the code to do this back in April, there were no conflicts with base branch and changes can be cleanly merged and made a pull request 2 months ago.

Without this critical feature, it essentially defeats the purpose for having this addon, since you have to scroll down EVERY time to see the most current message.

Can this please be implemented.

**Comment by humpfreybogart** [comment]
> [@Standard8](https://github.com/Standard8) status update on this? This has been requested for 4 years now. It appears that [Symplify62](https://github.com/Symplify62) added the code to do this back in April, there were no conflicts with base branch and changes can be cleanly merged and made a pull request 2 months ago.
> 
> Without this critical feature, it essentially defeats the purpose for having this addon, since you have to scroll down EVERY time to see the most current message.
> 
> Can this please be implemented.

I totally aggree - its really idiotic that this hasnt been implemented years ago. who the hell makes such a stupid decision to sort new mail in the bottom and not open the newest by default. 

come on!

**Comment by humpfreybogart** [comment]
in settings:

By default, expand: No messages, All messages, Automatic
add "Only latest message" 
and maybe add "Only latest message received" 

And add this setting :
Sort related messages by: Date ascending, Date descending

And on loading the thread related messages, just sort them. I can see the related messages is dynamically loaded so its just a matter of altering that code. show me where it is, and maybe I can do it.

Or as @godzfire writes, if someone already made a good solution, just implement it!

**Comment by Standard8** [comment]
> I totally aggree - its really idiotic that this hasnt been implemented years ago. who the hell makes such a stupid decision to sort new mail in the bottom and not open the newest by default.

At lot of people would say exactly the same if the original implementation was your preferred default. People are different and prefer different things.

However, it is never acceptable to attack people for the decisions they've made. I will not tolerate this and you will no longer be allowed to comment here.

I re-iterate that this is something I do on my own free time when I get around to it. I've been very busy in my personal life recently and I've only had the desire to keep the add-on going for now. When I feel like it, I might take a look at the PR that's been proposed, but I haven't really felt like it recently (and comments like this don't help).

**Comment by godzfire** [comment]
@Standard8 I think everyone expects to see their newest messages at the top of a convo. It just makes logical sense and the defacto standard for mail programs for decades, including Outlook.

And with respect, it's not fair to punish everyone as a whole and delay/not implement the change due to one person's comments. The code is there, ready to be utilized. Please implement it.

I truly believe it's the one thing stopping a lot of people from using this addon and keeping it, and once it's done, there will be a huge increase in usage; ideally maybe down the road finally push Mozilla group to natively implement it themselves, which should have been done ages ago.

**Comment by Standard8** [comment]
> [@Standard8](https://github.com/Standard8) I think everyone expects to see their newest messages at the top of a convo.

Please don't assume your way is ubiquitous unless you have actual statistics. I know for a fact that plenty of people over the years have said they are happy with the add-on without mentioning the sort order. The people who have complained about the sort order are in much smaller numbers. Normally the people that are unhappy complain the most.

> It just makes logical sense and the defacto standard for mail programs for decades, including Outlook.

To you it does, and to some other people, but not everyone.
 
> And with respect, it's not fair to punish everyone as a whole and delay/not implement the change due to one person's comments. The code is there, ready to be utilized. Please implement it.

If I'm being attacked for not doing things the correct way, how is it going to encourage me to spend time on it? I've already said that I haven't had much time recently. Whilst I do keep an eye on incoming messages, and occasional keep-it-going updates, that's about it.
 
> I truly believe it's the one thing stopping a lot of people from using this addon and keeping it, and once it's done, there will be a huge increase in usage;

You're welcome to your opinion, but please don't bother sharing it in future. I'll work on it when I want to. This is a hobby that I do in my free time when I feel like it. I do not get paid for it. Its something I do because I want to.
